pkg/azuredx/tokenprovider/token_provider_test.go

```diff
@@ -0,0 +1,102 @@
+package tokenprovider
+
+import (
+	"context"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+	"github.com/stretchr/testify/require"
+)
+
+const TestClientId = "3c882424-c381-45c0-a123-0a62ebc989dc"
+const TestTenantId = "b931e0dd-f42c-444a-83a9-63de6f2d771a"
+const TestClientSecret = "3093e953-91e1-48f4-9ad0-e8f750c38f93"
+const Scope = "https://kusto.kusto.windows.net/.default"
+const ExpectedToken string = "4cb83b87-0ffb-4abd-82f6-48a8c08afc53"
```

Comment:
just double checking we're cool with having all of these ids/secrets/etc be public? I guess I would think most tests would be with mocks, so it wouldn't matter what ids we used, but maybe that's not true.

Reply:
They're not actual credentials, I just generated the GUIDs online so that the have the format of credentials so it should be fine. But thanks for checking!